### PR TITLE
[Draft] Add SRLI and basic pattern matches for SRL, SRLI

### PR DESCRIPTION
**Pull request by hdelan**
Add support for SRLI instruction and basic DAG to DAG for SRL, SRLI.

### CODE REVIEW

File: llvm/lib/Target/StubV/StubVInstrInfo.td [hunk 1]
```diff
@@ -479,6 +479,37 @@ def SRL : Instruction {
   let mayStore = 0;
 }
 
+def SRLI : Instruction {
```
Comment (frasercrmck) [inline llvm/lib/Target/StubV/StubVInstrInfo.td]: Sorry, please rebase. I pushed a commit that simplifies the definitions of all these ALU instructions as it was getting a bit unwieldy. You should find `SRLI` much simpler to define.

**Review by frasercrmck — COMMENTED** [review]
Looking good but might be a good exercise to add tests.

**Comment by hdelan** [comment]
Thanks @frasercrmck tests added and have rebased!

**Review by frasercrmck — APPROVED** [review]
LGTM, let's shift it (right) up a gear!

**Comment by frasercrmck** [comment]
Since I accidentally merged it with the `Draft` tag, I was a bit naughty and force-pushed an amendment. Sorry!

**Pull request merged**